bin/docserv-createconfig

```diff
@@ -194,26 +200,73 @@ def switchbranch(repo):
         # change back to initial branch
         repo.checkout(initial_ref)
 
-def get_dc_files(path, branches, filter_dc=None):
-    """Collect all DC files in branches.
+def get_deliverables(path, branches, filter_dc=None):
+    """Collect all DC files and subdeliverables in all branches.
 
     :param str path: the path to the Git repository
     :param dict branches: A dictionary with lists as keys like
          {'branch1': [], 'branch2': [], ...}
     :param filter_dc: filter
-    :return: a dictionary with all DC files found in this branch
-    :rtype: dict
+    :return: a tuple with two dictionaries for each branch:
+        first dictionary contains all DC files
+        second dictionary contains all subdeliverables
+    :rtype: tuple(dict, dict)
     """
     repo = pygit2.Repository(path)
     dc_files = branches
+    subdeliverables = {}
     with switchbranch(repo) as checkout_branch:
         for branch in checkout_branch(branches):
             # some branches have their dc-files in a 'en' folder
             path2git = path + "/en" if "en" in os.listdir(path) else path
             # collecting dc-files in this branch
             dc_files[branch] = collect_dc_files(path2git, filter_dc)
-    return dc_files
+            # collecting subdeliverables for dc-all-files
+            dc_all_files = [item for item in dc_files[branch] if item.endswith("-all")]
+            for dc_all_file in dc_all_files:
+                main_all = get_main_rootid(path2git, dc_all_file)[0]
+                subdeliverables[branch + dc_all_file] = collect_subdeliverables(path2git, dc_files[branch], main_all)
+    return dc_files, subdeliverables
+
+def get_main_rootid(path, dc_file):
+    """Get the MAIN and ROOTID values from a DC-file
+
+    :param str path: the path to the Git repository
+    :param str dc_file: the DC-file to look in
+    :return: a tuple with two strings: MAIN and ROOTID vaules
+    :rtype: tuple(str, str)
+    """
+    main = rootid = ""
+    # example: ('MAIN', '=', '"', 'MAIN.susestorage.xml', '"')
+    #        ('ROOTID', '=', None, 'book.storage.admin', None)
+    regex = r'([A-Z]+)(=)("|\')?([\w\.\-]+)("|\')?'
```

Comment:
* There can be any number of spaces on either side of the `=` character.
* I think this does not correctly account for comment lines, you can just add `^` to the beginning of the regex, so it will only match at line start.

Reply:
>     * There can be any number of spaces on either side of the `=` character.

Are you sure?  That doesn't seem to work in bash. For example, if I enter in Bash:

```shell
$ a = 2
```

that gives me:

```
If 'a' is not a typo you can use command-not-found to lookup the package that contains it, like this:
    cnf a
```

Same in scripts ("command not found").

Reply:
>     * I think this does not correctly account for comment lines, you can just add `^` to the beginning of the regex, so it will only match at line start.

Your message inspired me to look a bit more closely to the regex. :wink: For example:

```python
regex = r'(?P<name>[A-Z]+)=(?:"|\')?(?P<value>[\w\.\-]+)(?:"|\')?'
```

The difference between the former regex:

* The equal sign `=` is a not a group anymore. The benefit is, it does not end up in the result anymore.
* The variable name and the variable value contains a name (`name` and `value`) now. That could make it a bit easier and more readable to access these two.
* The quote signs are now non-capturing groups, so they don't end up in the result.

In other words: the above regex lets you get only two entries: name and value (instead of four). Here is an example session:

```python
>>> re.search(regex, "A=555").groups()
('A', '555')
>>> re.search(regex, "A=555").groupdict()
{'name': 'A', 'value': '555'}
>>> re.search(regex, "A=555").group('name')
'A'
```

Adding the above regex you can improve the code and avoid the ugly `.group(4)` which is a bit hard to read:

```python
def get_main_rootid(path, dc_file):
    # ...
    regex = r'(?P<name>[A-Z]+)=(?:"|\')?(?P<value>[\w\.\-]+)(?:"|\')?'
    with open(os.path.join(path, dc_file), "r") as f:
        for line in f:
            match = re.match(regex, line)
            if match and match.group('name') == "MAIN":
                 main = match.group('value')
            if match and match.group('name') == "ROOTID":
                 rootid = match.group('value')
    return main, rootid
```

Reply:
> Are you sure? That doesn't seem to work in bash. For example, if I enter in Bash

Bash is irrelevant now, DAPS uses its own parser for DC files since v3.0 (or even since 2.4).

Reply:
> > Are you sure? That doesn't seem to work in bash. [...]
> 
> Bash is irrelevant now, DAPS uses its own parser for DC files since v3.0 (or even since 2.4). (That's why sourcing of DC file before running DAPS no longer works.)

Sorry for nitpicking, but DAPS allowed to source the DC file. With spaces, that wouldn't work anymore.

However, I'm not sure if this use case is still supported. I never used it.

Reply:
Thanks! Changes are done